### DISCUSSION TITLE
back and forward navigation in a puzzle

### DISCUSSION THREAD

**Discussion by arnaule** [rodpolako/Chess-PGN-Trainer#9] [Ideas]
After practicing with a few puzzles i feel it would be a great addition to be able to use the arrow keys to go back and forward inside a puzzle, to help memorize the line.
Thanks

**Comment by arnaule** [comment]
maybe this could go together with displaying the moves : 1e4..e5 2... on the screen, in notation in addition to the graphical view

**Comment by rodpolako** [comment]
Thanks for the suggestion.  In this particular case however, I don't think that it makes sense here because the ability to navigate within the puzzle is contrary to the design philosophy of the tool which is to drill as efficiently as possible.  Going back-and-forth in the puzzle would result in slower completion of the puzzles.

For puzzle move navigation, I think this is actually best accomplished by using the "Analysis board" option when running through the set.  Any puzzle you load with this option on will give you a link to an analysis board which will allow you to both do navigation as well as seeing the moves.  

Likewise, I'm trying to keep the visual distractions to a minimum and since there is the analysis board option showing the individual moves would add to the visual clutter.

**Comment by arnaule** [comment]
Thank you. For me the analysis button only appears in the "puzzle example", not in the others (anderssen_kieseritzky_1851 or in my files)

**Reply by rodpolako** [reply]
That would be correct because the "analysis board" feature only works with puzzles or positions with a FEN.  A full game would start at the beginning so there would be no FEN and therefore nothing to analyse.  If you want to analyse full games, use the analysis board feature of either chess.com or lichess.org and import the PGN there.

If you mean about the ability to auto-select the "Analysis board" feature when opening a PGN (such as your own files), please refer to the "Default settings via PGN Tags" section of the main Github page.  You have to add a [PGNTrainerAnalysisLink "1"] tag to the top of your file if you want the analysis board option selected every time you open the PGN.

**Comment by rodpolako** [comment]
Hi arnaule,
   I recently added annotations to the application which means now that the move numbers will show up on the left-hand side of the window (as long as you are not on a phone or a small screen).  I'm still not sure about the ability to move back-forth through a puzzle but it may be possible now that there is a manual next button available. I'll keep it mind.
